src/engine/strat_engine/thinpool.rs

```diff
@@ -31,6 +39,14 @@ pub struct ThinPool {
     thin_pool: ThinPoolDev,
     meta_spare: Vec<Segment>,
     id_gen: ThinDevIdPool,
+    filesystems: Table<StratFilesystem>,
+    mdv: MetadataVol,
+}
+
+pub enum ThinPoolStatus {
```

Comment:
This is supposed to be an overall status for the Stratis ThinPool struct. It feels like it ought to be a product type rather than a sum type, like:

```
struct ThinPoolStatus {
  thinpool_status: dm::ThinPoolStatus,
  ...other statuses
}
```